Makefile

```diff
@@ -0,0 +1,202 @@
+man7_src = $(filter-out $(programs) $(configs) topics/index.md,$(topics))
+
+# Target man pages
+man1     = $(man1_src:topics/%.md=$(MAN_DIR)/man1/valkey-%.1)
+man3     = $(man3_src:commands/%.md=$(MAN_DIR)/man3/%.3valkey)
+man4     = $(man4_src:topics/%.md=$(MAN_DIR)/man4/%.4)
+man7     = $(man7_src:topics/%.md=$(MAN_DIR)/man7/valkey-%.7) $(MAN_DIR)/man7/valkey-commands.7 $(MAN_DIR)/man7/valkey.7
+
+man_targets = $(man1) $(man3) $(man4) $(man7)
+
+$(man1): | $(MAN_DIR)/man1
+$(man3): | $(MAN_DIR)/man3
+$(man4): | $(MAN_DIR)/man4
+$(man7): | $(MAN_DIR)/man7
+$(MAN_DIR)/man1 $(MAN_DIR)/man3 $(MAN_DIR)/man4 $(MAN_DIR)/man7:
+	mkdir -p $@
+
+man_scripts = utils/preprocess-markdown.py utils/command_syntax.py utils/links-to-man.py
+
+$(MAN_DIR)/man1/valkey-%.1: topics/%.md $(man_scripts)
+	utils/preprocess-markdown.py --man --page-type program $< \
+	 | utils/links-to-man.py - | pandoc -s --to man -o $@ -
+$(MAN_DIR)/man3/%.3valkey: commands/%.md $(VALKEY_ROOT)/src/commands/%.json $(BUILD_DIR)/.commands-per-group.json $(man_scripts)
+	utils/preprocess-markdown.py --man --page-type command \
+	 --commands-per-group-json $(BUILD_DIR)/.commands-per-group.json \
+	 --valkey-root $(VALKEY_ROOT) $< \
+	 | utils/links-to-man.py - | pandoc -s --to man -o $@ -
+$(MAN_DIR)/man4/%.4: topics/%.md $(man_scripts)
+	utils/preprocess-markdown.py --man --page-type config $< \
+	 | utils/links-to-man.py - | pandoc -s --to man -o $@ -
+$(MAN_DIR)/man7/valkey-%.7: topics/%.md $(man_scripts)
+	utils/preprocess-markdown.py --man --page-type topic $< \
+	 | utils/links-to-man.py - | pandoc -s --to man -o $@ -
+$(MAN_DIR)/man7/valkey.7: topics/index.md $(man_scripts)
+	utils/preprocess-markdown.py --man --page-type topic $< \
+	 | utils/links-to-man.py - | pandoc -s --to man -o $@ -
+$(MAN_DIR)/man7/valkey-commands.7: $(BUILD_DIR)/.commands-per-group.json groups.json utils/build-command-index.py
+	utils/build-command-index.py --man \
+	 --groups-json groups.json \
+	 --commands-per-group-json $(BUILD_DIR)/.commands-per-group.json \
+	 | pandoc -s --to man -o $@ -
+
+.PHONY: install-man uninstall-man
+install-man: man | $(INSTALL_MAN_DIR)/man1 $(INSTALL_MAN_DIR)/man3 $(INSTALL_MAN_DIR)/man4 $(INSTALL_MAN_DIR)/man7
+	cp $(MAN_DIR)/man1/valkey*.1 $(INSTALL_MAN_DIR)/man1/
+	cp $(MAN_DIR)/man3/*.3valkey $(INSTALL_MAN_DIR)/man3/
+	cp $(MAN_DIR)/man4/valkey*.4 $(INSTALL_MAN_DIR)/man4/
+	cp $(MAN_DIR)/man7/valkey*.7 $(INSTALL_MAN_DIR)/man7/
+
+$(INSTALL_MAN_DIR)/man1 $(INSTALL_MAN_DIR)/man3 $(INSTALL_MAN_DIR)/man4 $(INSTALL_MAN_DIR)/man7:
+	mkdir -p $@
+
+uninstall-man:
+	rm $(INSTALL_MAN_DIR)/man1/valkey*.1
+	rm $(INSTALL_MAN_DIR)/man3/*.3valkey
+	rm $(INSTALL_MAN_DIR)/man4/valkey*.4
+	rm $(INSTALL_MAN_DIR)/man7/valkey*.7
+
+# -------
+
+# All targets
+targets = $(man_targets) $(html_targets) $(md_targets)
+
+md: $(md_targets)
+
+html: $(html_targets)
+
+man: $(man_targets)
+
+clean:
+	rm -f $(targets) $(BUILD_DIR)/.commands-per-group.json
```

Comment:
This appears to delete all the markdown files. Presumably because they are part of the make target.

Reply:
The md targets are pre-processed markdown files, which can be made using `make md`. They could be useful as an intermediate format for generating other formats, website, etc. or just debugging these scripts.

Anything in $(targets) are targets. The source files are in $(topics) and $(commands).

Reply:
Ok, let me go actually figure out why `make clean` was deleting all the .md files.

Reply:
Ok, I figured it out, it's actually the html targets which the culprit. I'll post in the right part.

Reply:
OOPS! You're right, it does!